apps/api/src/schema/library-traffic.ts

```diff
@@ -0,0 +1,43 @@
+import { z } from "@hono/zod-openapi";
+
+export const libraryTrafficQuerySchema = z.object({
+  locationName: z
+    .string()
+    .openapi({
+      example: "3rd Floor",
+      description: "Filter results by exact location name",
+    })
+    .optional(),
+});
+
+export const libraryTrafficEntrySchema = z.object({
+  id: z.number().int().nonnegative().openapi({ example: 245 }),
+
+  locationName: z.string().openapi({
+    example: "3rd Floor",
+    description: "Name of the library location",
+  }),
+
+  trafficCount: z.number().int().nonnegative().openapi({
+    example: 57,
+    description: "Number of people currently detected at the location",
+  }),
+
+  trafficPercentage: z
+    .union([z.string(), z.number().min(0).max(1)])
```

Comment:
this should be written `z.coerce.number()`

in fact, since this schema will handle your outgoing result, there should be no need to coerce at all since you are exactly certain of the type

note also that `.min` and `.max` will be enforced but are not visible on the openapi schema at all